charts/tidb-operator/templates/controller-manager-deployment.yaml

```diff
@@ -35,8 +35,8 @@ spec:
           - /usr/local/bin/tidb-controller-manager
           {{- if .Values.tidbBackupManagerImage }}
           - -tidb-backup-manager-image={{ .Values.tidbBackupManagerImage }}
-          - -tidb-discovery-image={{ .Values.operatorImage }}
           {{- end }}
+          - -tidb-discovery-image={{ .Values.operatorImage }}
```

Comment:
bugfix (it should not be in the above if statement)

Reply:
cc @onlymellb

Reply:
`.Values.tidbBackupManagerImage` should control `-tidb-backup-manager-image` flag only
`-tidb-discovery-image` must be set to current tidb-operator image, otherwise the default `tidb-operator:latest` will be always used which is apparently not right

Reply:
Good catch